src/main/java/cs2103/v15_1j/jimjim/model/Task.java

```diff
@@ -1,47 +1,74 @@
 package cs2103.v15_1j.jimjim.model;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
 
 import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.ObjectProperty;
 import javafx.beans.property.SimpleBooleanProperty;
-import javafx.beans.property.SimpleStringProperty;
-import javafx.beans.property.StringProperty;
+import javafx.beans.property.SimpleObjectProperty;
 
-public class Task {
+public class Task extends TaskEvent implements Comparable<Task> {
 
-	private final StringProperty taskName;
-	private final BooleanProperty completed;
+	private ObjectProperty<LocalDateTime> dateTime;
+	private ObjectProperty<LocalDate> date;
```

Comment:
Hmm. I don't like this extra variable date, as it can be created from dateTime.toLocalDate. if you want to separate date and time, use LocalDate and LocalTime